src/System.Private.CoreLib/shared/System/Math.cs

```diff
@@ -163,24 +165,51 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask
+                = unchecked((long)0b_1000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000__0000_0000);
 
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.IsSupported)
+            {
+                // Create vectors of the elements, and make them float to allow using SSE rather than SSE2 instructions
+                var xvec = Vector128.CreateScalarUnsafe(x).AsSingle();
+                var yvec = Vector128.CreateScalarUnsafe(y).AsSingle();
 
-            // If the sign bits of x and y are not the same,
-            // flip the sign bit of x and return the new value;
-            // otherwise, just return x
+                // Remove the sign from x, and remove everything but the sign from y
+                // Creating from a 'const long' is better than from the correct 'const float/double'
+                var mask = Vector128.CreateScalarUnsafe(signMask).AsSingle();
+                xvec = Sse.AndNot(mask, xvec);
+                yvec = Sse.And(mask, yvec);
```

Comment:
It's best to change the order of statements and parameters as below:
```C#
            yvec = Sse.And(yvec, mask);
            xvec = Sse.AndNot(mask, xvec);
```
Without VEX the current ordering generates results in some extra register moves:
```asm
       48B80000000000000080 mov      rax, 0x8000000000000000
       66480F6ED0           movd     xmm2, rax
       0F28DA               movaps   xmm3, xmm2
       0F54D9               andps    xmm3, xmm1
       0F28CB               movaps   xmm1, xmm3
       0F55D0               andnps   xmm2, xmm0
       0F28C2               movaps   xmm0, xmm2
       0F56C1               orps     xmm0, xmm1
```
With the proper ordering you get:
```asm
       48B80000000000000080 mov      rax, 0x8000000000000000
       66480F6ED0           movd     xmm2, rax
       0F54CA               andps    xmm1, xmm2
       0F55D0               andnps   xmm2, xmm0
       0F28C2               movaps   xmm0, xmm2
       0F56C1               orps     xmm0, xmm1
```

Reply:
Fixed